artemis-commons/src/main/java/org/apache/activemq/artemis/api/core/DeadLetterAddressRoutingType.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.api.core;
+
+public enum DeadLetterAddressRoutingType {
+
+   MULTICAST, ANYCAST, AS_ORIGIN;
+
+   public byte getType() {
+      switch (this) {
+         case MULTICAST:
+            return 0;
+         case ANYCAST:
+            return 1;
+         case AS_ORIGIN:
```

Comment:
I think we have something already on a third state somewhere else, where essentially the original is preserved we should probably reuse that enum or at least use same name for third case

Reply:
Third state might not be needed, could simply be represented as null i config, where when null the meaning is that it uses the parent setting

Reply:
I suppose AS_ORIGIN might be a misleading name.
What it does is that it makes sure messages are delivered to corresponding queues.
Imagine you have one address with 10 queues with multicast and consumption from q10 fails. So what it does it will deliver that failed message to DLA.q10 only, but not any others in DLA.
Same for anycast as this will ignore anycast routing and deliver to DLA.q10.
So it is not about taking routing from parent destination but rather providing predictive DL message delivery to corresponding DLQ instead of routed with ANYCAST or MULTICAST.

Reply:
THis is now removed.